.github/workflows/master.yml

```
@@ -0,0 +1,77 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2024 Intel Corporation
+
+name: Master workflow
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Build
+        run: |
+          go build -o build/_output/onos ./cmd/onos
+          go build -o build/_output/onos-cli-docs-gen ./cmd/onos-cli-docs-gen
+          go build -o build/_output/gnmi_cli ./cmd/gnmi_cli
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - uses: golangci/golangci-lint-action@v3.7.0
+        with:
+          version: latest
+          args: -v --config ./.golangci.yml
+
+  unit-tests:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Unit tests
+        run: |
+          go test github.com/onosproject/onos-cli/pkg/...
+          go test github.com/onosproject/onos-cli/cmd/...
+
+  docker-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+      - name: Build Docker image
+        run: |
+          go mod vendor
+          docker build . -f build/onos/Dockerfile \
+                  -t onosproject/onos-cli:latest
+          rm -rf vendor
```

Do you really need the `vendor`
```suggestion
          docker build . -f build/onos/Dockerfile \
                  -t onosproject/onos-cli:latest
```

---

But here you are building the Docker image, how come the test fails? would not running `go mod tidy` in the repo would resolve the need for `vendor`?

---

I don't think that will work because inside the Dockerfile we are coping ./vendor /build/vendor which will the test is failing. 

ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref 9368eea6-35a7-407f-8893-7ddd489a1d43::yjkt59pat04918msil9mc9e8e: "/vendor": not found

This is the error I'm facing without vendor

---

Ok, then let's leave it as is for now and we can revisit in the future